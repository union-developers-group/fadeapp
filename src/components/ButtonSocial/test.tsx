import { render, screen } from '@testing-library/react'

import { ButtonSocial, ButtonSocialProps } from '.'

const { getByRole, getByText } = screen

const data: ButtonSocialProps = {
  text: 'instagram',
  urlIcon: 'https://res.cloudinary.com/user/image/upload/img.png',
  urlRedirect: 'https://instagram.com/',
}

describe('<ButtonSocial />', () => {
  it('should render the deafult', () => {
    render(<ButtonSocial {...data} />)

    const text = getByText('instagram')
    const link = getByRole('link')
    const image = getByRole('img', { name: 'instagram' })

    expect(text).toBeInTheDocument()
    expect(link).toHaveAttribute('href', 'https://instagram.com/')
    expect(image).toHaveAttribute(
      'src',
      'https://res.cloudinary.com/user/image/upload/img.png',
    )
    expect(image).toHaveAttribute('alt', 'instagram')
  })

  it('Should match snapshot', () => {
    const { container } = render(<ButtonSocial {...data} />)

    expect(container).toMatchSnapshot()
  })
})
