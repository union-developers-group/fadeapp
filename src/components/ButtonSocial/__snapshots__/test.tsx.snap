// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonSocial /> Should match snapshot 1`] = `
<div>
  <div
    class="
flex 
flex-col 
items-center 
gap-4
"
  >
    <a
      class="
flex 
h-16 
w-16 
items-center 
justify-center 
rounded-full 
bg-primary 
transition 
ease-in-out 
hover:bg-primary/[0.6]
"
      href="https://instagram.com/"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="instagram"
        class="
max-w-[2.625rem]
"
        src="https://res.cloudinary.com/user/image/upload/img.png"
      />
    </a>
    <span
      class="
text-sm 
font-medium 
capitalize 
text-white
"
    >
      instagram
    </span>
  </div>
</div>
`;
